src/ReactIntl.re

```diff
@@ -124,9 +131,13 @@ module Intl = {
   [@bs.send] external now: (t, unit) => int = "";
 };
 
-type intl = Intl.t;
+[@bs.val] [@bs.module "react-intl"]
+external context: React.Context.t(Intl.t) = "IntlContext";
+
+// Not in react-intl yet
+// [@bs.module "react-intl"] external useIntl: unit => intl = "";
 
-[@bs.module "react-intl"] external useIntl: unit => intl = "";
+let useIntl = () => context->React.useContext; // not zero-cost but will be when react-intl will add it
```

Comment:
👍 I am not sure they will add it, because they have React 16.3 as their minimum version now, and hooks were only added in 16.8.

Reply:
I hope at least in the future versions.